packages/vkui/src/components/CustomSelect/CustomSelect.test.tsx

```diff
@@ -550,8 +555,265 @@ describe('CustomSelect', () => {
       />,
     );
 
+    expect(onChange).toBeCalledTimes(0);
     expect(getCustomSelectValue()).toEqual('Mike');
     fireEvent.click(screen.getByRole('button', { hidden: true }));
     expect(getCustomSelectValue()).toEqual('');
+
+    expect(onChange).toBeCalledTimes(1);
+  });
+
+  it('(uncontrolled): calls onChange when click on unselected option and does not call when click on selected ', async () => {
+    const onChange = jest.fn((event: React.ChangeEvent<HTMLSelectElement>) => event.target.value);
+
+    render(
+      <CustomSelect
+        data-testid="target"
+        options={[
+          { value: 0, label: 'Mike' },
+          { value: 1, label: 'Josh' },
+        ]}
+        allowClearButton
+        onChange={onChange}
+        defaultValue={0}
+      />,
+    );
+
+    expect(onChange).toBeCalledTimes(0);
+    expect(getCustomSelectValue()).toEqual('Mike');
+
+    fireEvent.click(screen.getByTestId('target'));
+    expect(screen.getByTitle('Mike')).toHaveAttribute('aria-selected', 'true');
+    fireEvent.mouseEnter(screen.getByTitle('Josh'));
+    fireEvent.click(screen.getByTitle('Josh'));
```

Comment:
```suggestion
    expect(screen.getByRole('option', { selected: true })).toHaveTextContent('Mike');
    const elUnselectedOption = screen.getByRole('option', { name: 'Josh', selected: false });
    fireEvent.mouseEnter(elUnselectedOption);
    fireEvent.click(elUnselectedOption);
```
давай заменим в тестах на такие селекторы, это сразу и на доступность проверит

- дока по методу `getByRole` https://testing-library.com/docs/queries/byrole/

Reply:
Отличная идея 🏃

Reply:
@inomdzhon ⚡ https://github.com/VKCOM/VKUI/pull/5123/commits/b2b1d61d83368d7883e56c87caf46b4e2e3b9147